flang-rt/lib/runtime/character.cpp

```diff
@@ -293,6 +293,17 @@ inline RT_API_ATTRS std::size_t Index(const CHAR *x, std::size_t xLen,
     }
     return 0;
   }
+  if (wantLen == 1) {
+    // Trivial case for single character lookup.
+    // We can use simple forward search.
+    CHAR ch = want[0];
```

Comment:
This could just become a call to `std::memchr()` for KIND==1.

For other kinds, please add braced initialization to the two declarations here.